### PR TITLE
Fix specs related to loading models

### PR DESCRIPTION
**Pull request by aviflax**
…which was merged earlier this week via #23.

I was reviewing some of this code while resolving some merge conflicts resulting from rebasing a local branch onto master, and I noticed that a few things were off in some of the specs in `src/fc4/dsl/model.clj` and `src/fc4/io/dsl.clj`.

## Notes to reviewers

* I recommend reviewing each commit individually.
* Thank you!

### CODE REVIEW

File: src/fc4/dsl/model.clj [hunk 1]
```diff
@@ -223,11 +223,15 @@
     (catch ParserException e
       (fault (str "YAML could not be parsed: error " e)))))
 
+;; NB: this is used in at least one other namespace.
```
Comment (AndreaCrotti) [inline src/fc4/dsl/model.clj]: Just to understand, why do you need this NB?

File: test/fc4/io/dsl_test.clj [hunk 2]
```diff
@@ -1,70 +1,102 @@
...
             [fc4.util             :as u])
   (:import [java.io FileNotFoundException]))
 
 (u/namespaces '[fc4 :as f]
               '[fc4.model :as m])
 
+;; At first I just called instrument with no args, but I ran into trouble with some specs/fns inside
+;; clj-chrome-devtools. So I came up with this overwrought approach to instrumenting only the functions in
+;; the namespace under test.
+(->> (ns-interns 'fc4.io.dsl)
```
Comment (AndreaCrotti) [inline test/fc4/io/dsl_test.clj]: Probably for a separete PR but If you use Kaocha as test runner you don't need any of this, it does the instrumentation for you.
In general I think that instrumenting in each test file never quite works, since if you work with others someone will forget and some specs will not be checked.

File: src/fc4/dsl/model.clj (hunk 1 above)
Comment (aviflax) [reply src/fc4/dsl/model.clj]: Well, specs are often defined in a namespace and then used only in that namespace — i.e. by code in the same file. In this case, we have a spec in a different file that’s using this one, so I wanted to warn future maintainers so they’d be aware of this uncommon dependency before changing or removing this spec.

Comment (AndreaCrotti) [reply src/fc4/dsl/model.clj]: Well that should be caught by tests anyway, and I guess they would do a simple search before removing a spec?
Anyway we can leave the NB I was just curious to know why.

File: test/fc4/io/dsl_test.clj (hunk 2 above)
Comment (aviflax) [reply test/fc4/io/dsl_test.clj]: Thanks, I agree. I have a comment somewhere else about moving this stuff to the project’s test-runner-runner. (One of the reasons I haven’t done so is because I’m also considering just deleting the test-runner-runner and using Kaocha instead.) (I’m stalling on Kaocha because it doesn’t support parallelism when running tests, and this test suite is heavy on computation due to the heavy usage of generative property testing, so I _think_ it benefits substantially from the parallelism support in eftest.)

Comment (AndreaCrotti) [reply test/fc4/io/dsl_test.clj]: Parallelism can still be achieved easily by marking tests in different ways (unit/integration etc) with different jobs in CircleCI.
Is not exactly the same thing but you get the same result in the end imho.

File: src/fc4/dsl/model.clj (hunk 1 above)
Comment (aviflax) [reply src/fc4/dsl/model.clj]: 👍 🙏

File: test/fc4/io/dsl_test.clj (hunk 2 above)
Comment (aviflax) [reply test/fc4/io/dsl_test.clj]: That makes sense for CI, and seems like a decent solution (if a bit contrived) but it wouldn’t help when running the entire test suite locally… although perhaps that’s an edge case? I suppose I do that less and less frequently these days; I more commonly run a specific test namespace(s) for the namespace(s) I’m working on, then once _those_ pass I’ll push to GH so CI will run the full test suite for me… 🤔

Comment (aviflax) [reply test/fc4/io/dsl_test.clj]: Thanks for spurring my thinking along!
